src/internal/connector/discovery/api/users.go

```diff
@@ -165,6 +166,77 @@ func (c Users) GetByID(ctx context.Context, identifier string) (models.Userable,
 	return resp, err
 }
 
+func (c Users) GetUserInfo(
+	ctx context.Context,
+	userID string,
+) (
+	userPurpose string,
+	hasMailBox, hasOneDrive bool,
+	err error,
+) {
+	var (
+		rawURL  = "https://graph.microsoft.com/v1.0/users/" + userID + "/mailboxSettings"
+		apadtor = c.stable.Adapter()
+		builder = users.NewUserItemRequestBuilder(rawURL, apadtor)
+	)
+
+	// verify mailbox enabled for user
+	_, err = c.stable.Client().UsersById(userID).MailFolders().Get(ctx, nil)
+	hasMailBox = true
+
+	if err != nil {
+		if !graph.IsErrExchangeMailFolderNotFound(err) {
+			return "", false, false, graph.Wrap(ctx, err, "error getting mail folder")
+		}
+
+		logger.Ctx(ctx).Infof("resource owner does not have a mailbox enabled")
+
+		hasMailBox = false
+	}
+
+	// verify onedrive enabled for user
+	_, err = c.stable.Client().UsersById(userID).Drives().Get(ctx, nil)
+	hasOneDrive = true
+
+	if err != nil {
+		err = graph.Stack(ctx, err)
+		if !clues.HasLabel(err, graph.LabelsMysiteNotFound) {
+			return "", false, false, graph.Wrap(ctx, err, "error getting onedrive")
+		}
+
+		logger.Ctx(ctx).Infof("resource owner does not have a drive")
+
+		hasOneDrive = false
+	}
+
+	// get userPurpose for user
+	newItem, err := builder.Get(ctx, nil)
+	if err == nil {
+		userPurpose = ptr.Val(newItem.GetAdditionalData()["userPurpose"].(*string))
+	}
+
+	if err != nil {
+		if !(graph.IsErrAccessDenied(err) || graph.IsErrExchangeMailFolderNotFound(err)) {
+			return "", false, false, graph.Wrap(ctx, err, "getting purpose")
+		}
+
+		// not the best way to handle the errors. But since we don't won't to break
+		// this request and just want to convey a message that access are invalid have
```

Comment:
I had a discussion with @vkamra that since its a new change we don't won't to break things for existing users with lesser permissions. 
Can we add one more value here specifying err if any. And that goes as part of user's struct output?

Reply:
I don't have the full context of the issue, but is "user" a reasonable fallback in case of an error?

Reply:
No. its just that we can have different type of users like - user or shared.

Reply:
Returning a value as an ignored-error indicator is a smell.  First and foremost, we need to ensure there's an error log in all of these cases.  That will do most of the work in communicating the occurrence.  If we need to pivot logic based on this state, then we'll want a better set of identifiers than a descriptive string, anyway (passing up an error and letting a higher layer handle it, for example).